code/server/test/api_endpoint_tests.js

```diff
@@ -1,26 +1,29 @@
 var app = require('../main.js');
 var request = require('supertest');
 var assert = require('assertthat'); // View README for documentation https://github.com/thenativeweb/assertthat
-
+var api = require('../routes/api.js');
+var stubUser = require('../stub_models/StubUser.js');
+var stubLikes = require('../stub_models/StubUserMatches.js');
 //////////////////////////////////////////
 // ALL API TESTS SHOULD GO IN THIS FILE // 
 //////////////////////////////////////////
 
 describe('api', function() {
+          api.injectUser(stubUser);
+          api.injectLikes(stubLikes);
```

Comment:
I read into it, looks like when you require a file that is already in the cache, node know to reference that same file instead of creating a new instance. So in this case you are actually referencing the same api file that was created by main.

So that's why it works, I'm not sure if this follows best practices, but at least we know now.